packages/assets-controllers/src/TokensController.ts

```diff
@@ -115,6 +121,33 @@ export interface TokensState extends BaseState {
   suggestedAssets: SuggestedAssetMeta[];
 }
 
+/**
+ * The name of the {@link TokensController}.
+ */
+const controllerName = 'TokensController';
+
+/**
+ * The external actions available to the {@link TokensController}.
+ */
+type AllowedActions =
+  | AddApprovalRequest
+  | AcceptApprovalRequest
+  | RejectApprovalRequest;
+
+/**
+ * The messenger of the {@link TokensController}.
+ */
+export type TokensControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AllowedActions,
+  never,
+  AllowedActions['type'],
+  never
+>;
+
+const ORIGIN_METAMASK = 'metamask';
+const WATCH_ASSET_METHOD_NAME = 'wallet_watchAssets';
```

Comment:
This can use the `APPROVAL_TYPE` constant once it exists, but for now would it be more explicit to call it `APPROVAL_TYPE`?